### PR TITLE
Use applied rule in warnSimplificationWithRemainder

### LINKED ISSUE #1570: Warning for simplification with remainders should report which rule was applied

### PR DESCRIPTION
**Pull request by andreiburdusa**
---
Fixes #1570 
###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

File: kore/src/Kore/Step/Axiom/Registry.hs [hunk 1]
```diff
@@ -162,10 +162,10 @@ equalitiesToEvaluators
   =
```
Comment (andreiburdusa) [inline kore/src/Kore/Step/Axiom/Registry.hs]: This function takes multiple rules, and `scatter`s them, so I try to figure out how to keep track of each rule

File: kore/src/Kore/Log/WarnSimplificationWithRemainder.hs [hunk 2]
```diff
@@ -102,26 +107,24 @@ warnSimplificationWithRemainder
     => InternalVariable variable
     => TermLike variable  -- ^ input pattern
     -> SideCondition variable  -- ^ input condition
-    -> OrPattern variable  -- ^ results
     -> OrPattern variable  -- ^ remainders
+    -> EqualityRule Variable
```
Comment (ana-pantilie) [inline kore/src/Kore/Log/WarnSimplificationWithRemainder.hs]: I think you can remove the TODO above now.

File: kore/src/Kore/Step/Axiom/EvaluationStrategy.hs [hunk 3]
```diff
@@ -101,8 +101,20 @@ simplificationEvaluation rule =
     BuiltinAndAxiomSimplifier $ \term condition -> do
         results' <- evaluateAxioms [rule] condition term
         let initial = Step.toConfigurationVariables (Pattern.fromTermLike term)
+            remainders' = Results.remainders results'
         Step.recoveryFunctionLikeResults initial results'
-        return $ Results.toAttemptedAxiom results'
+        let attemptedAxiom = Results.toAttemptedAxiom results'
+        case attemptedAxiom of
```
Comment (ana-pantilie) [inline kore/src/Kore/Step/Axiom/EvaluationStrategy.hs]: I think you can factor this `case` out to make the code more readable.

Comment (ttuegel) [reply kore/src/Kore/Step/Axiom/EvaluationStrategy.hs]: I agree, this should be something like `when (hasRemainder attemptedAxiom) $ warnSimplificationWithRemainder ...`

**Pull request merged**